src/PointwiseFunctions/AnalyticSolutions/Elasticity/HalfSpaceMirror.hpp

```diff
@@ -0,0 +1,182 @@
+    static constexpr OptionString help{
+        "Relative tolerance for numerical integrals"};
+    static type lower_bound() noexcept { return 0.; }
+    static type upper_bound() noexcept { return 1.; }
+    static type default_value() noexcept { return 1e-10; }
+  };
+
+  using options = tmpl::list<BeamWidth, Material, IntegrationIntervals,
+                             AbsoluteTolerance, RelativeTolerance>;
+  static constexpr OptionString help{
+      "A semi-infinite mirror on which a laser introduces stress perpendicular "
+      "to the mirrors surface. The displacement then is the Hankel-Transform "
+      "of the general solution multiplied by the beam profile"};
+
+  HalfSpaceMirror() = default;
+  HalfSpaceMirror(const HalfSpaceMirror&) noexcept = delete;
+  HalfSpaceMirror& operator=(const HalfSpaceMirror&) noexcept = delete;
+  HalfSpaceMirror(HalfSpaceMirror&&) noexcept = default;
+  HalfSpaceMirror& operator=(HalfSpaceMirror&&) noexcept = default;
+  ~HalfSpaceMirror() noexcept = default;
+
+  HalfSpaceMirror(double beam_width,
+                  constitutive_relation_type constitutive_relation,
+                  size_t no_intervals = 350, double absolute_tolerance = 1e-12,
+                  double relative_tolerance = 1e-10) noexcept;
+
+  const constitutive_relation_type& constitutive_relation() const noexcept {
+    return constitutive_relation_;
+  }
+
+  // @{
+  /// Retrieve variable at coordinates `x`
+  auto variables(const tnsr::I<DataVector, 3>& x,
+                 tmpl::list<Tags::Displacement<3>> /*meta*/) const noexcept
+      -> tuples::TaggedTuple<Tags::Displacement<3>>;
+
+  auto variables(const tnsr::I<DataVector, 3>& x,
+                 tmpl::list<Tags::Strain<3>> /*meta*/) const noexcept
+      -> tuples::TaggedTuple<Tags::Strain<3>>;
+
+  static auto variables(
+      const tnsr::I<DataVector, 3>& x,
+      tmpl::list<::Tags::FixedSource<Tags::Displacement<3>>> /*meta*/) noexcept
+      -> tuples::TaggedTuple<::Tags::FixedSource<Tags::Displacement<3>>>;
+  // @}
+
+  /// Retrieve a collection of variables at coordinates `x`
+  template <typename... Tags>
+  tuples::TaggedTuple<Tags...> variables(const tnsr::I<DataVector, 3>& x,
+                                         tmpl::list<Tags...> /*meta*/) const
+      noexcept {
+    static_assert(sizeof...(Tags) > 1, "An unsupported Tag was requested.");
+    return {tuples::get<Tags>(variables(x, tmpl::list<Tags>{}))...};
+  }
+
+  // clang-tidy: no pass by reference
+  void pup(PUP::er& p) noexcept;  // NOLINT
+
+ private:
+  friend bool operator==(const HalfSpaceMirror& lhs,
+                         const HalfSpaceMirror& rhs) noexcept;
+
+  double beam_width_{std::numeric_limits<double>::signaling_NaN()};
+  constitutive_relation_type constitutive_relation_{};
+  size_t no_intervals_{};
+  double absolute_tolerance_{};
+  double relative_tolerance_{};
+};
+
+bool operator!=(const HalfSpaceMirror& lhs,
+                const HalfSpaceMirror& rhs) noexcept;
```

Comment:
Why not inline != here?

Reply:
Why would this operator be an exception to the preference of defining functions in the source file to reduce compile times?